test/test_array_ops.cpp

```diff
@@ -498,6 +498,29 @@ TEST(ArrayOps, OneHot3dNegAxis) {
 
 }  // end of op OneHot
 
+TEST(ArrayOps, Pad) {
+  Scope root = Scope::NewRootScope();
+
+  vector<int> static_input_indexes = {1};
+
+  Tensor input(DT_INT32, TensorShape({2, 3}));
```

Comment:
Can you check a 3D case as well, with input rank = 3 and paddings of size 3x2

Reply:
Also, I think in the 2D case, paddings of size 1x2 is a valid input. I think Pad op is able to broadcast it up. We might not have the broadcast in the translate function right now

Reply:
Also it seems that TF can pad a tensor of max rank = 6. Try a 7D tensor (padding = 7x2) and see if it fails. Do we throw the same error that TF throws. In the same spirit, can you please check if we are running TF's unit tests for Pad and passing them all?

Reply:
It seems TF pad is able to take a 3rd input sometimes, which contains the value to be padded with.
Right now we only pass 0 to ngraph. In the translate op please read in this input and pass it along to ngraph, and add a unit for non-zero pad value

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/pad_op.cc#L74

Reply:
The padding input seems to have to be positive numbers only. Please throw the error TF might throw if we encounter negative numbers

Reply:
The above corner cases are based off a cursory reading of the asserts in the code of Pad's Compute here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/pad_op.cc#L56

Please take a look at them and see if there are any other cases we miss out on supporting or testing

Reply:
Filed separate ticket to handle corner cases of Pad: https://jira.devtools.intel.com/browse/NGTF-1952